### PR TITLE
Remove code that is not needed anymore due the changes of how extendW…

### PR DESCRIPTION
**Pull request by normanmaurer**
…ith(...) works now

Motivation:

We can remove some code-duplication now as CompositeBuffer.extendWith(...) now takes care of all the extra logic

Modifications:

Remove not needed code

Result:

Cleanup

Motivation:

Explain here the context, and why you're making that change.
What is the problem you're trying to solve.

Modification:

Describe the modifications you've done.

Result:

Fixes #<GitHub issue number>. 

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

File: codec/src/test/java/io/netty5/handler/codec/compression/CompressionTestUtils.java [hunk 1]
```diff
@@ -64,15 +64,13 @@ static CompositeBuffer compose(BufferAllocator allocator, Supplier<Buffer> suppl
...
-                    // Trim off the excess with split.
-                    bufferList.add(msg.split().send());
-                } else {
+
+                // This is needed at the moment as otherwise we will see:
+                //
+                // java.lang.AssertionError: Method should not be used.
+                // at io.netty5.buffer.api.DefaultCompositeBuffer$TornBufferAccessor
+                //     .readByte(DefaultCompositeBuffer.java:1575)
+                if (msg.readableBytes() != 0) {
```
Comment (normanmaurer) [inline codec/src/test/java/io/netty5/handler/codec/compression/CompressionTestUtils.java:73]: @chrisvest I didn't have time to debug this further but this seems like a bug in `DefaultCompositeBuffer`

**Comment by normanmaurer** [comment]
@chrisvest want to merge this or wait for you to investigate first ?

**Comment by chrisvest** [comment]
@normanmaurer We can merge this first, as the problem isn't introduced here.

**Comment by normanmaurer** [comment]
ok let me do it ...

**Pull request merged**